presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSink.java

```diff
@@ -92,15 +98,28 @@ public RaptorPageSink(
         this.sortFields = ImmutableList.copyOf(sortColumnIds.stream().map(columnIds::indexOf).collect(toList()));
         this.sortOrders = ImmutableList.copyOf(requireNonNull(sortOrders, "sortOrders is null"));
 
+        this.bucketCount = bucketCount;
         this.bucketFields = bucketColumnIds.stream().mapToInt(columnIds::indexOf).toArray();
 
-        this.pageWriter = bucketCount.isPresent() ? new BucketedPageWriter(bucketCount.getAsInt()) : new SimplePageWriter();
-
         for (int field : bucketFields) {
             if (!columnTypes.get(field).equals(BIGINT)) {
                 throw new PrestoException(NOT_SUPPORTED, "Bucketing is only supported for BIGINT columns");
             }
         }
+
+        requireNonNull(temporalColumnHandle, "temporalColumnHandle is null");
```

Comment:
This is not needed, since you use it on the next line (null check only matters if you are storing it for later)